--- 本地实验/SYNC-BARRIER-ELISION-X/V002/parent.asc	2026-10-06 23:04:43.411373821 +0000
+++ 本地实验/SYNC-BARRIER-ELISION-X/V002/submission.asc	2026-10-06 23:13:35.245254551 +0000
@@ -590,7 +590,6 @@
                 FromFloat(outputLocal, valueLocal, valid);
                 AscendC::PipeBarrier<PIPE_V>();
                 AscendC::Mul(outputLocal, outputLocal, gammaLocal, valid);
-                AscendC::PipeBarrier<PIPE_V>();
                 AscendC::Add(outputLocal, outputLocal, biasLocal, valid);
                 AscendC::PipeBarrier<PIPE_V>();
                 SyncVToMTE3();
