--- a/submission.asc
+++ b/submission.asc
@@ -556,7 +556,6 @@
             }
             AscendC::PipeBarrier<PIPE_V>();
             AscendC::Mul(xFp32, valueLocal, valueLocal, valid);
-            AscendC::PipeBarrier<PIPE_V>();
             AscendC::ReduceSum(reduceLocal, xFp32, residualFp32, valid);
             AscendC::SetFlag<AscendC::HardEvent::V_MTE2>(inputRelease);
             needInputRelease = true;
