diff --git a/phase4/online/R31A/V016/submission.asc b/phase4/local/R31A/V021/submission.asc
index dec9ae5..1896e11 100644
--- a/phase4/online/R31A/V016/submission.asc
+++ b/phase4/local/R31A/V021/submission.asc
@@ -2051,8 +2051,8 @@ private:
                 SyncVToMTE2();
                 SyncVToMTE3();
                 Store(outputGm_, rowOffset + col, valueLocal[col], valid);
-                SyncMTE3ToV();
             }
+            SyncMTE3ToV();
         }
     }
 
